Release 5.0.0

bastelfreak opened the pull request.

bastelfreak commented:
I would like to include https://github.com/voxpupuli/rspec-puppet-facts/pull/189 as well

ekohl commented on CHANGELOG.md:
```diff
@@ -1,5 +1,19 @@
+
+[Full Changelog](https://github.com/voxpupuli/rspec-puppet-facts/compare/4.0.0...5.0.0)
+
+for a long long time, the first level of keys in a factsets were symbols. Following nested hashes always had strings. This doesn't make sense and looks confusing. Also the original data contains only strings, we we actually iterated on the data and converted it to symbols. We now flip the default. This saves performance and makes code more readable. This major release also switched to FacterDB 3, ...
```
```suggestion
For a long long time, the first level of keys in a factsets were symbols. This was fine before there were structured facts, but structured facts ended up as nested hashes that always had strings. This doesn't make sense and easy to get wrong. The original data contains only strings so conversion actually cost performance.

We now flip the default to increase performance, make code more readable and remove some footguns. This major release also switched to FacterDB 3, which doesn't contain [legacy facts](https://www.puppet.com/docs/puppet/7/core_facts.html) nor factset for EoL operating systems or EoL facter versions.
```

ekohl reviewed (changes requested):
This suggestion assumes https://github.com/voxpupuli/rspec-puppet-facts/pull/189 is merged, but I'm a bit hesitant on that. I'd prefer to formally deprecate it first.

That means:
* Make sure the configuration option to switch is well documented
* Migration guide
* Announce the next major version will drop it
* Communicate this (also to PDK users)

bastelfreak commented:
* I raised https://github.com/voxpupuli/rspec-puppet-facts/pull/193 which updates the README.md and deprecates symbolized facts
* I updated the release notes for 3.0.0 (https://github.com/voxpupuli/rspec-puppet-facts/issues/170), say mentioned that symbolized keys are deprecated with the 5.0.0 release
* I moved https://github.com/voxpupuli/rspec-puppet-facts/issues/190 to the release summary for 6.0.0
